### PR TITLE
Twilson63/chore setup repository 1

### PR DESCRIPTION
**Pull request by twilson63**
In this PR, I setup the repo to build for both Deno and NodeJS `cjs` and `esm` styles, also setup typescript checking for both NodeJS and Deno. My plan is just to start with tests in Deno, hopefully I can keep this whole codebase in both deno and NodeJS

Please review and approve when you have a chance

🙏🏻

### CODE REVIEW

File: egg.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "$schema": "https://x.nest.land/eggs@0.3.8/src/schema.json",
+  "name": "hyper-ext-model",
+  "entry": "./src/mod.ts",
```
Comment (TillaTheHun0) [inline egg.json:4]: Should this be `./dist/bundle.js`?

**Review by TillaTheHun0 — COMMENTED** [review]
Just one question. Other than that, looks good.

File: egg.json (hunk 1 above)
Comment (twilson63) [reply egg.json]: Yep, good catch!

**Review by TillaTheHun0 — APPROVED** [review]
LGTM!

**Pull request merged**